components/mediation-admin/org.wso2.carbon.mediation.library/src/main/java/org/wso2/carbon/mediation/library/service/MediationLibraryAdminService.java

```diff
@@ -144,11 +144,19 @@ public LibraryInfo getLibraryInfo(String libName, String packageName) throws Axi
 		SynapseImport synImport = new SynapseImport();
 		synImport.setLibName(libName);
 		synImport.setLibPackage(packageName);
+		SynapseConfiguration configuration = getSynapseConfiguration();
+		SynapseImport synaImport = configuration.getSynapseImports().get("{"+packageName+"}"+libName);
 		OMElement impEl = SynapseImportSerializer.serializeImport(synImport);
 		if (impEl != null) {
 			try {
 				OMElement imprtElem = createElement(impEl.toString());
-				SynapseImport synapseImport = SynapseImportFactory.createImport(imprtElem, null);
+				SynapseImport synapseImport = null;
+				if(synaImport == null) {
```

Comment:
Have only one variable. no need to keep two. Just synaImport will do. Please revise your code accordinly